lib/src/top_level.rs

```diff
@@ -754,20 +812,35 @@ impl TopLevelItem {
                 args,
                 ret_ty,
                 body,
+                is_dead_code,
             } => fn_to_doc(
                 name,
                 Some(ty_params),
                 Some(where_predicates),
                 args,
                 ret_ty,
                 body,
+                *is_dead_code,
             ),
-            TopLevelItem::Module { name, body } => group([
-                nest([text("Module"), line(), text(name), text(".")]),
-                nest([hardline(), body.to_doc()]),
-                hardline(),
-                nest([text("End"), line(), text(name), text(".")]),
-            ]),
+            TopLevelItem::Module {
+                name,
+                body,
+                is_dead_code,
+            } => {
+                let mut dead_code_attribute = String::new();
+                if *is_dead_code {
+                    dead_code_attribute
+                        .push_str("(* #[allow(dead_code)] - module was ignored by the compiler *)");
+                    dead_code_attribute.push('\n');
+                }
+                group([
+                    nest([text(dead_code_attribute)]),
```

Comment:
```suggestion
                    if is_dead_code {
                      text("(* #[allow(dead_code)] - module was ignored by the compiler *)")
                    } else {
                      nil()
                    },
```
to avoid the mutation and the additional `nest` (not needed for a single element)